### ISSUE TITLE
do not reproduce the yolox-nano result 

### ISSUE THREAD

**Issue by Tim5Tang** [Megvii-BaseDetection/YOLOX#674]
**use the cmd and  parameter as follow:**

python tools/train.py -n yolox-nano -d 2 -b 64 --fp16 -o --cache

2021-09-08 14:42:06.344 | INFO     | yolox.core.trainer:before_train:126 - args: Namespace(batch_size=64, cache=True, ckpt=None, devices=1, dist_backend='nccl', dist_url=None, exp_file=None, experiment_name='nano', fp16=True, machine_rank=0, name='yolox-nano', num_machines=1, occupy=True, opts=[], resume=False, start_epoch=None)
2021-09-08 14:42:06.347 | INFO     | yolox.core.trainer:before_train:127 - exp value:
╒══════════════════╤════════════════════════════╕
│ keys             │ values                     │
╞══════════════════╪════════════════════════════╡
│ seed             │ None                       │
├──────────────────┼────────────────────────────┤
│ output_dir       │ './YOLOX_outputs'          │
├──────────────────┼────────────────────────────┤
│ print_interval   │ 10                         │
├──────────────────┼────────────────────────────┤
│ eval_interval    │ 10                         │
├──────────────────┼────────────────────────────┤
│ num_classes      │ 80                         │
├──────────────────┼────────────────────────────┤
│ depth            │ 0.33                       │
├──────────────────┼────────────────────────────┤
│ width            │ 0.25                       │
├──────────────────┼────────────────────────────┤
│ data_num_workers │ 4                          │
├──────────────────┼────────────────────────────┤
│ input_size       │ (416, 416)                 │
├──────────────────┼────────────────────────────┤
│ multiscale_range │ 5                          │
├──────────────────┼────────────────────────────┤
│ data_dir         │ None                       │
├──────────────────┼────────────────────────────┤
│ train_ann        │ 'instances_train2017.json' │
├──────────────────┼────────────────────────────┤
│ val_ann          │ 'instances_val2017.json'   │
├──────────────────┼────────────────────────────┤
│ mosaic_prob      │ 0.5                        │
├──────────────────┼────────────────────────────┤
│ mixup_prob       │ 1.0                        │
├──────────────────┼────────────────────────────┤
│ hsv_prob         │ 1.0                        │
├──────────────────┼────────────────────────────┤
│ flip_prob        │ 0.5                        │
├──────────────────┼────────────────────────────┤
│ degrees          │ 10.0                       │
├──────────────────┼────────────────────────────┤
│ translate        │ 0.1                        │
├──────────────────┼────────────────────────────┤
│ mosaic_scale     │ (0.5, 1.5)                 │
├──────────────────┼────────────────────────────┤
│ mixup_scale      │ (0.5, 1.5)                 │
├──────────────────┼────────────────────────────┤
│ shear            │ 2.0                        │
├──────────────────┼────────────────────────────┤
│ perspective      │ 0.0                        │
├──────────────────┼────────────────────────────┤
│ enable_mixup     │ False                      │
├──────────────────┼────────────────────────────┤
│ warmup_epochs    │ 5                          │
├──────────────────┼────────────────────────────┤
│ max_epoch        │ 300                        │
├──────────────────┼────────────────────────────┤
│ warmup_lr        │ 0                          │
├──────────────────┼────────────────────────────┤
│ basic_lr_per_img │ 0.00015625                 │
├──────────────────┼────────────────────────────┤
│ scheduler        │ 'yoloxwarmcos'             │
├──────────────────┼────────────────────────────┤
│ no_aug_epochs    │ 15                         │
├──────────────────┼────────────────────────────┤
│ min_lr_ratio     │ 0.05                       │
├──────────────────┼────────────────────────────┤
│ ema              │ True                       │
├──────────────────┼────────────────────────────┤
│ weight_decay     │ 0.0005                     │
├──────────────────┼────────────────────────────┤
│ momentum         │ 0.9                        │
├──────────────────┼────────────────────────────┤
│ exp_name         │ 'nano'                     │
├──────────────────┼────────────────────────────┤
│ test_size        │ (416, 416)                 │
├──────────────────┼────────────────────────────┤
│ test_conf        │ 0.01                       │
├──────────────────┼────────────────────────────┤
│ nmsthre          │ 0.65                       │
├──────────────────┼────────────────────────────┤
│ random_size      │ (10, 20)                   │
╘══════════════════╧════════════════════════════╛
2021-09-08 14:42:06.468 | INFO     | yolox.core.trainer:before_train:133 - Model Summary: Params: 0.91M, Gflops: 1.08
2021-09-08 14:42:08.182 | INFO     | yolox.data.datasets.coco:__init__:45 - loading annotations into memory...
2021-09-08 14:42:20.787 | INFO     | yolox.data.datasets.coco:__init__:45 - Done (t=12.60s)
2021-09-08 14:42:20.787 | INFO     | pycocotools.coco:__init__:92 - creating index...
2021-09-08 14:42:21.736 | INFO     | pycocotools.coco:__init__:92 - index created!
2021-09-08 14:42:50.371 | WARNING  | yolox.data.datasets.coco:_cache_images:69 - 

**but the result is :**

2021-09-10 05:46:06.608 | INFO     | yolox.core.trainer:save_ckpt:318 - Save weights to ./YOLOX_outputs/nano
2021-09-10 05:46:16.802 | INFO     | yolox.evaluators.coco_evaluator:evaluate_prediction:171 - Evaluate in main process...
2021-09-10 05:46:20.158 | INFO     | yolox.evaluators.coco_evaluator:evaluate_prediction:204 - Loading and preparing results...
2021-09-10 05:46:24.156 | INFO     | yolox.evaluators.coco_evaluator:evaluate_prediction:204 - DONE (t=4.00s)
2021-09-10 05:46:24.156 | INFO     | pycocotools.coco:loadRes:433 - creating index...
2021-09-10 05:46:24.287 | INFO     | pycocotools.coco:loadRes:433 - index created!
2021-09-10 05:46:36.375 | INFO     | yolox.core.trainer:evaluate_and_save_model:309 - 
Average forward time: 0.52 ms, Average NMS time: 0.56 ms, Average inference time: 1.08 ms
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.238
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.388
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.246
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.074
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.249
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.383
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.228
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.351
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.375
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.128
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.407
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.579

2021-09-10 05:46:36.376 | INFO     | yolox.core.trainer:save_ckpt:318 - Save weights to ./YOLOX_outputs/nano
2021-09-10 05:46:36.418 | INFO     | yolox.core.trainer:after_train:184 - Training of experiment is done and the best AP is 23.83

**Comment by Joker316701882** [comment]
@Tim5Tang With this version, we can get the best mAP around 24.

However, the results reported in our tech report and this repo are based on an ImageNet pre-trained model. That's why you can't reproduce the result. But noted that yolox_nano is the only one that adopts the pretrain model.

**Comment by Tim5Tang** [comment]
ok, i see thanks.
